README.md

```diff
@@ -24,88 +22,117 @@ Via CDN:
 
-A callback function must be passed in for the redirect flows(loginRedirect and acquireTokenRedirect). This callback function is called after the authentication request is completed either successfully or with a failure. This is not required for the popup flows since they return promises.
-
-UserAgentApplication has other optional parameters like redirectUri which can be assigned. Please refer to the [Wiki](https://github.com/AzureAD/microsoft-authentication-library-for-js/wiki/MSAL-basics#configuration-options) to see the full list and their default values.
+After instantiating your instance, if you plan on using a redirect flow (`loginRedirect` and `acquireTokenRedirect`), you must register a callback handlers  using `handleRedirectCallbacks(successHandler, errorHandler)`. The callback function is called after the authentication request is completed either successfully or with a failure. This is not required for the popup flows since they return promises
 
 ```JavaScript
-    var applicationConfig = {
-        clientID: 'your_client_id'
+    var msalConfig = {
+        auth: {
+            clientId: 'your_client_id'
+        }
     };
 
-     var userAgentApplication = new Msal.UserAgentApplication(applicationConfig.clientID, null, tokenReceivedCallback);
+    var msalInstance = new Msal.UserAgentApplication(msalConfig);
+
+    msalInstance.handleRedirectCallbacks(response => {
+        // handle redirect response
+    }), err => {
+       // handle error
+    });
 
-    //callback function for redirect flows
-    function tokenReceivedCallback(errorDesc, token, error, tokenType) {
-        if (token) {
-        }
-        else {
-            log(error + ":" + errorDesc);
-        }
-    }
 ```
 
 #### 2. Login the user
 
-Your app must login the user with either loginPopup or the loginRedirect method to establish user context. When the login methods are called and the authentication of the user is completed by the Azure AD service, an [id token](https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens) is returned which is used to identify the user with some basic information.
+Your app must login the user with either the `loginPopup` or the `loginRedirect` method to establish user context.
+
+When the login methods are called and the authentication of the user is completed by the Azure AD service, an [id token](https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens) is returned which is used to identify the user with some basic information.
+
+When you login a user, you can request scopes up front if you know they will be needed, but it is not required, and login can be called with no params. It is best practice to only request scopes you need when you need them, a concept called dynamic consent. AAD will only allow you to get consent for 3 resources at a time, although you can request many scopes within a resource.
```

Comment:
This text is a little confusing for usage. I am adding a detailed text here - may be you can summarize and add a link on scopes usage in login/acquire calls etc and expand it in the link:

When you login a user, and do not know what scopes you would need eventually, you can call the login() APIs without any scopes. For usecases where you may want the user to pre-consent for resource access later, you can pass the scopes during login. 
    - Please note that the passing of scopes at login allows you to request consent for the scopes, but the token returned will not contain the scopes. 
    - The user still would be calling acquireToken() for the 'token' with scopes. However this can be a non-interactive silent call if the user has preconsented the scopes in the previous login call.

It is usually best practice to have 'dynamic consent' app model. If your app model fits in 'dynamic consent' which is where you only request scopes you need when you need them, any number of acquireToken() calls can be made per need basis. Please expect interactive consent screen for those calls whose scopes are not preconsented.

AAD will only allow you to get consent for 3 resources at a time, although you can request many scopes within a resource. In this context:
    - When the user makes a login request, multiple resources and their corresponding scopes can be passed since the service issues an idToken preconsenting those scopes
    - However acquireToken calls are valid only for one resource/multiple scopes. If you have a usecase of accessing multiple resources, please split the acquireToken calls one per resource

Reply:
When you login a user, you can pass in scopes that the user can pre consent to on login, however this is not required. Please note that consenting to scopes on login, does not return an access_token for these scopes, but gives you the opportunity to obtain a token silently with these scopes passed in, with no further interaction from the user.  

It is best practice to only request scopes you need when you need them, a concept called dynamic consent. While this can create more interactive consent for users in your application, it also reduces drop-off from users that may be uneasy granting a large list of permissions for features they are not yet using.

AAD will only allow you to get consent for 3 resources at a time, although you can request many scopes within a resource.
When the user makes a login request, you can pass in multiple resources and their corresponding scopes because AAD issues an idToken pre consenting those scopes. However acquireToken calls are valid only for one resource / multiple scopes. If you need to access multiple resources, please make separate acquireToken calls per resource.

Reply:
incorporated feedback